src/test/java/io/datareplication/model/snapshot/SnapshotIndexTest.java

```diff
@@ -1,15 +1,202 @@
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.time.Instant;
+import java.time.format.DateTimeParseException;
 import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 class SnapshotIndexTest {
+
+    public static final String SCHEMA_URL_FOR_SNAPSHOT_INDEX =
+        "https://www.datareplication.io/spec/snapshot/index.schema.json";
+    private final ObjectMapper mapper = new ObjectMapper();
+    private JsonSchema schema;
+
+    @BeforeEach
+    void setUp() throws URISyntaxException {
+        schema = fetchJsonSchemaFromUrl();
+    }
+
+    @Test
+    void schemaValidation_providedJsonValidates() throws IOException {
+        JsonNode node = getJsonNodeFromStringContent("{\n"
+                                                     + "    \"id\": \"12345678\",\n"
+                                                     + "    \"createdAt\": \"2023-09-27T20:52:17.000Z\",\n"
+                                                     + "    \"pages\": [\n"
+                                                     + "      \"https://example.datareplication.io/12345678/page/1\",\n"
+                                                     + "      \"https://example.datareplication.io/12345678/page/2\",\n"
+                                                     + "      \"https://example.datareplication.io/12345678/page/3\"\n"
+                                                     + "    ]\n"
+                                                     + "  }");
+        Set<ValidationMessage> errors = schema.validate(node);
+        assertThat(errors).isEmpty();
+    }
+
+    @Test
+    void schemaValidation_providedJsonFailsToValidateBecauseItHasMissingKeys() throws IOException {
+        JsonNode node = getJsonNodeFromStringContent("{\n"
+                                                     + "    \"id1\": \"12345678\",\n"
+                                                     + "    \"createdAt2\": \"2023-09-27T20:52:17.000Z\",\n"
+                                                     + "    \"pages3\": [\n"
+                                                     + "      \"https://example.datareplication.io/12345678/page/1\",\n"
+                                                     + "      \"https://example.datareplication.io/12345678/page/2\",\n"
+                                                     + "      \"https://example.datareplication.io/12345678/page/3\"\n"
+                                                     + "    ]\n"
+                                                     + "  }");
+        Set<ValidationMessage> errors = schema.validate(node);
+        assertThat(errors).hasSize(3);
+        assertThat(errors.stream().map(ValidationMessage::toString)).contains("$.id: is missing but it is required");
```

Comment:
`assertThat(errors.stream().map(ValidationMessage::getMessage))`

or alternative with custom build assertThat

`assertThat(errors).containsMessage("$.id: is missing but it is required")`

And for all following assertions